app/src/main/java/com/jojjnator/kalkylator/viewmodel/CalculatorViewModel.kt

```diff
@@ -4,58 +4,84 @@ import androidx.compose.runtime.mutableStateOf
 import androidx.lifecycle.ViewModel
 import com.jojjnator.kalkylator.CalculatorAction
 
-class CalculatorViewModel : ViewModel() {
 
-    // TODO: Send events from UI to viewModel.
-    // Event for numbers pressed
-    // Event for clear pressed
-    // Event for operations
-    // Event for decimal
-    // Event for doing the calculation
+class CalculatorViewModel : ViewModel() {
 
-    var state = mutableStateOf("")
+    var secondInput = mutableStateOf("")
         private set
-
-    var state2 = mutableStateOf("")
+    var firstInput = mutableStateOf("")
         private set
-
     var operator = mutableStateOf("")
         private set
 
-    fun checkInput(action: CalculatorAction) {
-        when (action) {
+    var calculatedSum = mutableStateOf(0.0)
+        private set
+
+    fun checkInput(actionFromUI: CalculatorAction) {
+        when (actionFromUI) {
             is CalculatorAction.Clear -> {
-                state.value = ""
-                state2.value = ""
                 operator.value = ""
+                secondInput.value = ""
+                firstInput.value = ""
+                calculatedSum.value = 0.0
             }
             is CalculatorAction.DoCalculation -> {
-                if (operator.value == "+") {
-
-                    state.value.toInt() + state2.value.toInt()
-                }
-                if (operator.value == "-") {
-                    state.value.toInt() - state2.value.toInt()
-                }
-                if (operator.value == "*") {
-                    state.value.toInt() * state2.value.toInt()
-                }
-                if (operator.value == "/") {
-                    state.value.toDouble() / state2.value.toDouble()
-                }
+                doCalculation(operatorInput = operator.value)
             }
             is CalculatorAction.Comma -> {
```

Comment:
These conditions could be one liners:

when (actionFromUI) {
    is CalculatorAction.Comma -> addCommaToDigit()
}

Reply:
Otherwise, I think, it looks great